src/Storages/System/StorageSystemStackTrace.cpp

```diff
@@ -214,6 +219,39 @@ ThreadIdToName getFilteredThreadNames(ASTPtr query, ContextPtr context, const Pa
     return tid_to_name;
 }
 
+bool parseHexNumber(std::string_view sv, UInt64 & res)
+{
+    errno = 0; /// Functions strto* don't clear errno.
+    char * pos_integer = const_cast<char *>(sv.begin());
+    res = std::strtoull(sv.begin(), &pos_integer, 16);
+    return (pos_integer == sv.begin() + sv.size() && errno != ERANGE);
+}
+bool isSignalBlocked(UInt64 tid, int signal)
```

Comment:
Ok. Is race-condition (on tid) possible? If it is the case, we need at least an explanation in comments.

Reply:
Yes, but this will not break anything. Two possible options in this case:
- it will incorrectly detect that the signal **is blocked** and will not send it this time
- it will incorrectly detect that the signal **is not blocked** then it will wait `storage_system_stack_trace_pipe_read_timeout_ms`

But anyway from my perspective it is better then returning incorrect stack races like now.

Reply:
And this is not the only place in system.stack_trace, reading thread names is also race, so you may report incorrect name

Reply:
But there is one possible issue right now - ignore ENOENT (like thread names reading does), I've addressed this, and added few comments here and there.